src/ciphersuite/mod.rs

```diff
@@ -76,6 +76,13 @@ impl Secret {
     pub(crate) fn to_vec(&self) -> Vec<u8> {
         self.value.clone()
     }
+
+    /// Randomly sample a fresh `Secret`.
+    pub(crate) fn from_random(length: usize) -> Self {
```

Comment:
I'm not sure what I should do here. The `value` field is a `Vec` and I think it has to be, because we can't say which length it's going to be at compile time. Should it have an additional `length` field?

Reply:
There are different ways to do this. One would be to have different `Secret` types for different lengths we know we have. The other option is to use something like [generic array](https://docs.rs/generic-array/0.14.4/generic_array/index.html) (or re-implement the minimal set of logic from there that we actually need here).
But you can file an issue for this and leave it open here. It might end up not being a small change.

Reply:
I'll file an issue. I think the consensus was that we wanted a `Secret` trait anyway.